packages/dds/tree/src/test/utils.ts

```diff
@@ -372,21 +372,42 @@ export class TestTreeProvider {
 	}
 }
 
-export interface ConnectionSetter {
-	readonly setConnected: (connectionState: boolean) => void;
-}
-
-export type SharedTreeWithConnectionStateSetter = ISharedTree & ConnectionSetter;
+/**
+ * A type with subset of functionalities of mock container runtime that can help tests control
+ * the processing of messages and the connection state of the runtime.
+ *
+ * - "connected": Set the connection state of a given tree. Any messages that are submitted while the tree is
+ * disconnected will be resubmitted when it reconnects by calling the resubmit function on the tree. Also, the
+ * clientId of the runtime changes on reconnection.
+ *
+ * - "pauseProcessing":  Pauses the processing of messages for a tree. Unlike during reconnection, messages that
+ * are submitted while the tree is paused will not be resubmitted when it unpauses.
```

Comment:
The current phrasing makes it sound as though the messages will never be submitted. Something like "Messages that are submitted while the tree is paused will be submitted when it unpauses (without going thought the resubmit flow)" would be clearer.
Also, I think it would be helpful to make separate statements about inbound and outbound messages.
The resubmit flow only makes sense in the context of outbound messages.

Same for "connected"

Reply:
Good point. The distinction between inbound and outbound is interesting, I hadn't thought about it from that angle. I updated the comments to be more specific and clear.